tempesta_fw/cfg.c

```diff
@@ -888,6 +888,73 @@ tfw_cfg_map_enum(const TfwCfgEnum mappings[],
 EXPORT_SYMBOL(tfw_cfg_map_enum);
 
 /**
+ * Get value of attribute with name @attr_key.
+ * Return @default_val if the attribute is not found in the entry @e.
+ */
+const char *
+tfw_cfg_get_attr(const TfwCfgEntry *e, const char *attr_key,
+		 const char *default_val)
+{
+	const char *key, *val;
+	size_t i;
+
+	TFW_CFG_ENTRY_FOR_EACH_ATTR(e, i, key, val) {
+		if (!strcasecmp(key, attr_key))
+			return val;
+	}
+
+	return default_val;
+}
+EXPORT_SYMBOL(tfw_cfg_get_attr);
+
+/**
+ * Check that integer is in specified range.
+ * Print an error and return non-zero if it is out of range.
+ */
+int
+tfw_cfg_check_range(long value, long min, long max)
+{
+	if (min != max && (value < min || value > max)) {
+		TFW_ERR("the value %ld is out of range: [%ld, %ld]\n",
```

Comment:
Agree with Alexey. What 'the value 20 is out of range: [0, 10]' will say in logs? What if we have many items with the value '20' in configuration? Rather we should see something like 'the value of feature_foo (20) is out of range: [0, 10]', so now we know that feature_foo is poorly configured.